### PR TITLE
improve composition query performance

### LINKED ISSUE #257: Improve composition
For a field of type reference, currently API looks up every record in the relevant collection

Considering a collection with X fields of type reference, an API query for N records will turn into N * X + 1 DB queries.
In the case of multiple references, this grows exponentially.

However, the _ids for each referenced collection could be grouped into a single query, and composition could happen in memory, bringing down the total number of queries to X + 1, regardless of the multiplicity of the references.

### PR DESCRIPTION
**Pull request by jimlambie**
Close #257 

This modifies the composer to perform a lookup of all ids listed in the array of documents it is passed, performing the composition in memory once all the results are in.

Previously, the database was queried for each id individually.

### CODE REVIEW

File: dadi/lib/composer/index.js [hunk 1]
```diff
@@ -15,114 +13,118 @@ Composer.prototype.setApiVersion = function (apiVersion) {
...
-      }
+      queue.push(
+        new Promise((resolve, reject) => {
+          var f = field
 
-      // are specific fields required?
-      var fields = {}
-      var schemaFields = help.getFromObj(this.model.schema, key + '.settings.fields', [])
+          model.find(query, { 'compose': compose, 'fields': fields }, (err, result) => {
+            if (err) console.log(err)
```
Comment (eduardoboucas) [inline dadi/lib/composer/index.js]: Should we `reject()` here?

File: test/acceptance/hooks.js [hunk 2]
```diff
@@ -15,20 +15,21 @@ var request = require('supertest')
 var bearerToken
 var connectionString = 'http://' + config.get('server.host') + ':' + config.get('server.port')
 
-describe('Hooks', function () {
+describe.skip('Hooks', function () {
```
Comment (eduardoboucas) [inline test/acceptance/hooks.js]: Don't forget to remove this before releasing 😅 💣 💀

File: dadi/lib/composer/index.js (hunk 1 above)
Comment (jimlambie) [inline dadi/lib/composer/index.js]: we probably ought to

File: test/acceptance/hooks.js (hunk 2 above)
Comment (jimlambie) [inline test/acceptance/hooks.js]: that's there for a reason... I'll try to remember the reason

**Pull request merged**